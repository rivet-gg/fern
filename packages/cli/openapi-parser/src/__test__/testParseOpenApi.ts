import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
