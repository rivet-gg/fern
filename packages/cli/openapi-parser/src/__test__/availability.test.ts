import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("availability", "openapi.yml");
});
