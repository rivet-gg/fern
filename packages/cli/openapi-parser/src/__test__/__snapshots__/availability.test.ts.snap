// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser availability parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [
            {
              "name": "id",
              "value": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "id",
                },
              },
            },
            {
              "name": "active_id",
              "value": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "active_id",
                },
              },
            },
            {
              "name": "deprecated_ref_id",
              "value": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "deprecated_ref_id",
                },
              },
            },
            {
              "name": "x-fern-availability_path_param",
              "value": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "x-fern-availability_path_param",
                },
              },
            },
          ],
          "queryParameters": [],
          "request": null,
          "response": {
            "properties": {
              "fine_setting": {
                "type": "primitive",
                "value": {
                  "type": "boolean",
                  "value": true,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequest",
      "headers": [
        {
          "availability": "Deprecated",
          "description": "deprecated header",
          "env": null,
          "name": "X-Deprecated-Header",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXDeprecatedHeader",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXDeprecatedHeader",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "availability": null,
          "description": "active header",
          "env": null,
          "name": "X-Active-Header",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXActiveHeader",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXActiveHeader",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "availability": null,
          "description": null,
          "env": null,
          "name": "X-Deprecated-Ref-Header",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXDeprecatedRefHeader",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXDeprecatedRefHeader",
              "groupName": null,
              "nameOverride": null,
              "schema": "DeprecatedParam",
              "type": "reference",
            },
          },
        },
        {
          "availability": "Beta",
          "description": "active beta header",
          "env": null,
          "name": "x-fern-availability_header",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXFernAvailabilityHeader",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXFernAvailabilityHeader",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": null,
      "pagination": null,
      "path": "/collection/{id}/{active_id}/{deprecated_ref_id}/{x-fern-availability_path_param}",
      "pathParameters": [
        {
          "availability": "Deprecated",
          "description": "collection id",
          "name": "id",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestId",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
          "variableReference": null,
        },
        {
          "availability": null,
          "description": "active header",
          "name": "active_id",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestActiveId",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
          "variableReference": null,
        },
        {
          "availability": null,
          "description": null,
          "name": "deprecated_ref_id",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestDeprecatedRefId",
            "groupName": null,
            "nameOverride": null,
            "schema": "DeprecatedParam",
            "type": "reference",
          },
          "variableReference": null,
        },
        {
          "availability": "Beta",
          "description": "active beta header",
          "name": "x-fern-availability_path_param",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXFernAvailabilityPathParam",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
          "variableReference": null,
        },
      ],
      "queryParameters": [
        {
          "availability": "Deprecated",
          "description": "language code",
          "name": "deprecated_lang",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestDeprecatedLang",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestDeprecatedLang",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "availability": null,
          "description": "language code",
          "name": "active_lang",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestActiveLang",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestActiveLang",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "availability": null,
          "description": null,
          "name": "deprecated_ref_lang",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestDeprecatedRefLang",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestDeprecatedRefLang",
              "groupName": null,
              "nameOverride": null,
              "schema": "DeprecatedParam",
              "type": "reference",
            },
          },
        },
        {
          "availability": "Beta",
          "description": "active beta query param",
          "name": "x-fern-availability_query_param",
          "parameterNameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXFernAvailabilityQueryParam",
            "groupName": null,
            "nameOverride": null,
            "type": "nullable",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamRequestXFernAvailabilityQueryParam",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "request": null,
      "requestNameOverride": null,
      "response": {
        "description": "OK",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "GetCollectionIdActiveIdDeprecatedRefIdXFernAvailabilityPathParamResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "WholeDeprecatedSettings",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [],
    },
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": {
            "properties": {},
            "type": "object",
          },
          "response": {
            "properties": {
              "not_deprecated_setting": {
                "type": "primitive",
                "value": {
                  "type": "boolean",
                  "value": true,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "PostSuccessRefRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "POST",
      "operationId": null,
      "pagination": null,
      "path": "/success-ref",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "PostSuccessRefRequest",
          "groupName": null,
          "nameOverride": null,
          "schema": "Settings",
          "type": "reference",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "OK",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "PostSuccessRefResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "NotDeprecatedSettings",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [],
    },
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": {
            "properties": {},
            "type": "object",
          },
          "response": {
            "properties": {
              "not_deprecated_setting": {
                "type": "primitive",
                "value": {
                  "type": "boolean",
                  "value": true,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "PostSuccessInlinePropertyDeprecationRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "POST",
      "operationId": null,
      "pagination": null,
      "path": "/success-inline-property-deprecation",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "additionalProperties": false,
          "allOf": [],
          "allOfPropertyConflicts": [],
          "availability": null,
          "description": null,
          "generatedName": "PostSuccessInlinePropertyDeprecationRequest",
          "groupName": null,
          "nameOverride": null,
          "properties": [
            {
              "audiences": [],
              "availability": "Deprecated",
              "conflict": {},
              "generatedName": "postSuccessInlinePropertyDeprecationRequestDeprecatedSetting",
              "key": "deprecated_setting",
              "nameOverride": null,
              "schema": {
                "availability": "Deprecated",
                "description": null,
                "generatedName": "postSuccessInlinePropertyDeprecationRequestDeprecatedSetting",
                "groupName": null,
                "nameOverride": null,
                "type": "optional",
                "value": {
                  "availability": "Deprecated",
                  "description": null,
                  "generatedName": "PostSuccessInlinePropertyDeprecationRequestDeprecatedSetting",
                  "groupName": null,
                  "nameOverride": null,
                  "schema": {
                    "default": null,
                    "type": "boolean",
                  },
                  "type": "primitive",
                },
              },
            },
            {
              "audiences": [],
              "availability": null,
              "conflict": {},
              "generatedName": "postSuccessInlinePropertyDeprecationRequestNewSetting",
              "key": "new_setting",
              "nameOverride": null,
              "schema": {
                "availability": null,
                "description": null,
                "generatedName": "postSuccessInlinePropertyDeprecationRequestNewSetting",
                "groupName": null,
                "nameOverride": null,
                "type": "optional",
                "value": {
                  "availability": null,
                  "description": null,
                  "generatedName": "PostSuccessInlinePropertyDeprecationRequestNewSetting",
                  "groupName": null,
                  "nameOverride": null,
                  "schema": {
                    "default": null,
                    "type": "boolean",
                  },
                  "type": "primitive",
                },
              },
            },
          ],
          "type": "object",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "OK",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "PostSuccessInlinePropertyDeprecationResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "NotDeprecatedSettings",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [],
    },
    {
      "audiences": [],
      "authed": false,
      "availability": "Deprecated",
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": {
            "properties": {},
            "type": "object",
          },
          "response": {
            "properties": {
              "not_deprecated_setting": {
                "type": "primitive",
                "value": {
                  "type": "boolean",
                  "value": true,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "PostSuccessInlineObjectDeprecationRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "POST",
      "operationId": null,
      "pagination": null,
      "path": "/success-inline-object-deprecation",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "additionalProperties": false,
          "allOf": [],
          "allOfPropertyConflicts": [],
          "availability": "Deprecated",
          "description": null,
          "generatedName": "PostSuccessInlineObjectDeprecationRequest",
          "groupName": null,
          "nameOverride": null,
          "properties": [
            {
              "audiences": [],
              "availability": null,
              "conflict": {},
              "generatedName": "postSuccessInlineObjectDeprecationRequestFineSetting",
              "key": "fine_setting",
              "nameOverride": null,
              "schema": {
                "availability": null,
                "description": null,
                "generatedName": "postSuccessInlineObjectDeprecationRequestFineSetting",
                "groupName": null,
                "nameOverride": null,
                "type": "optional",
                "value": {
                  "availability": null,
                  "description": null,
                  "generatedName": "PostSuccessInlineObjectDeprecationRequestFineSetting",
                  "groupName": null,
                  "nameOverride": null,
                  "schema": {
                    "default": null,
                    "type": "boolean",
                  },
                  "type": "primitive",
                },
              },
            },
          ],
          "type": "object",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "OK",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "PostSuccessInlineObjectDeprecationResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "NotDeprecatedSettings",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [],
    },
    {
      "audiences": [],
      "authed": false,
      "availability": "Beta",
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": {
            "properties": {},
            "type": "object",
          },
          "response": {
            "properties": {
              "not_deprecated_setting": {
                "type": "primitive",
                "value": {
                  "type": "boolean",
                  "value": true,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "PostSuccessInlineBetaBodyRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "POST",
      "operationId": null,
      "pagination": null,
      "path": "/success-inline-beta-body",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "additionalProperties": false,
          "allOf": [],
          "allOfPropertyConflicts": [],
          "availability": "Beta",
          "description": null,
          "generatedName": "PostSuccessInlineBetaBodyRequest",
          "groupName": null,
          "nameOverride": null,
          "properties": [
            {
              "audiences": [],
              "availability": null,
              "conflict": {},
              "generatedName": "postSuccessInlineBetaBodyRequestFineSetting",
              "key": "fine_setting",
              "nameOverride": null,
              "schema": {
                "availability": null,
                "description": null,
                "generatedName": "postSuccessInlineBetaBodyRequestFineSetting",
                "groupName": null,
                "nameOverride": null,
                "type": "optional",
                "value": {
                  "availability": null,
                  "description": null,
                  "generatedName": "PostSuccessInlineBetaBodyRequestFineSetting",
                  "groupName": null,
                  "nameOverride": null,
                  "schema": {
                    "default": null,
                    "type": "boolean",
                  },
                  "type": "primitive",
                },
              },
            },
          ],
          "type": "object",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "OK",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "PostSuccessInlineBetaBodyResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "NotDeprecatedSettings",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": null,
      "server": [],
      "summary": null,
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "DeprecatedParam",
    "WholeDeprecatedSettings",
    "NotDeprecatedSettings",
  ],
  "schemas": {
    "DeprecatedParam": {
      "availability": "Deprecated",
      "description": null,
      "generatedName": "DeprecatedParam",
      "groupName": null,
      "nameOverride": null,
      "schema": {
        "default": null,
        "format": null,
        "maxLength": null,
        "minLength": null,
        "pattern": null,
        "type": "string",
      },
      "type": "primitive",
    },
    "NotDeprecatedSettings": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "NotDeprecatedSettings",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "notDeprecatedSettingsNotDeprecatedSetting",
          "key": "not_deprecated_setting",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "notDeprecatedSettingsNotDeprecatedSetting",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "NotDeprecatedSettingsNotDeprecatedSetting",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "type": "boolean",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "Settings": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Settings",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": "Deprecated",
          "conflict": {},
          "generatedName": "settingsDeprecatedSetting",
          "key": "deprecated_setting",
          "nameOverride": null,
          "schema": {
            "availability": "Deprecated",
            "description": null,
            "generatedName": "settingsDeprecatedSetting",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": "Deprecated",
              "description": null,
              "generatedName": "SettingsDeprecatedSetting",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "type": "boolean",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "settingsNewSetting",
          "key": "new_setting",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "settingsNewSetting",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "SettingsNewSetting",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "type": "boolean",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "WholeDeprecatedSettings": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": "Deprecated",
      "description": null,
      "generatedName": "WholeDeprecatedSettings",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "wholeDeprecatedSettingsFineSetting",
          "key": "fine_setting",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "wholeDeprecatedSettingsFineSetting",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "WholeDeprecatedSettingsFineSetting",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "type": "boolean",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Deprecated params and fields",
  "variables": {},
  "webhooks": [],
}
`;
